packages/core/src/odata/v4/date-filter-function.ts

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { EntityBase } from '../common/entity';
+import {
+  Filter,
+  FilterFunction,
+  FilterFunctionParameterType
+} from '../common/filter';
+
+/**
+ * Representation of a filter function, that returns a value of type date. This supports DateTimeOffset values.
+ */
+export class DateFilterFunction<
```

Comment:
This looks very much like the `NumberFilterFunction` only the `edmType` seems to be different. Perhaps I overlooking something. If not would it be possible introduce one abstract layer from `FilterFunction` called `OrderableFilterFunction` or something indicating that you have the [<,>,=<,>=] methods on this filter?

Reply:
Thats true, although I am not sure this would be worth the effort and maybe it makes it more complicated than necessary. Anyways, if we decide for something like this, we should also apply this to `StringFilterFunction` and `BooleanFilterFunction` and then also for all the `Field` implementations that we have, because this is the case everywhere there. I would rather see this as out of scope for this PR.

Reply:
But StringFilterFunction and BooleanFilterFunction have no methods defined. Just the edmType if different which is fine for me.

I am bothered by the four comparing methods which are there two times in the Date and Number FilterFunction definition. I agree on the fileds but this would be really out of scope of this PR :-) but that there are even bigger code duplication in the existing code base does not justify to keep adding more in my opinion. 

I agree though that it is always a trade off of making things N times explicit and easy to understand or make it one time abstract but harder to understand. In our case N=2 I have no strong opinion. Are there other data types coming up increasing N to higher values?

Reply:
Hmm, not for this one, I will probably also add Collection types, but they also only support eq and ne.
I can add an additional class in between.

Reply:
Done, I think we should ideally use mixins here in the future and reuse those for the fields as well. But for this has to suffice 😊